framework/Steps.py

```diff
@@ -465,20 +474,31 @@ def _localTakeAstepRun(self,inDictionary):
     inputs     = inDictionary['Input'     ]
     outputs    = inDictionary['Output'    ]
     sampler    = inDictionary[self.splType]
+    # check to make sure model can be run
```

Comment:
Possibly not, but at some point these two approaches have to be coincided since they're mostly duplicated.  I think it's safer to keep it in both places.